python/ray/serve/_private/proxy_state.py

```diff
@@ -808,3 +761,25 @@ def _stop_proxies_if_needed(self) -> bool:
             proxy_state = self._proxy_states.pop(node_id)
             self._proxy_restart_counts[node_id] = proxy_state.proxy_restart_count + 1
             proxy_state.shutdown()
+
+
+def _try_cancel_future(fut: asyncio.Future, e: Exception):
+    if not fut.done():
+        fut.set_exception(e)
+
+
+def wrap_as_future(ref: ObjectRef, timeout_s: float) -> asyncio.Future:
+    loop = asyncio.get_running_loop()
+
+    aio_fut = asyncio.wrap_future(ref.future(), loop=loop)
+    # Schedule cancellation for the future
+    timeout_handler = loop.call_later(
+        max(timeout_s, 0),
+        _try_cancel_future,
+        aio_fut,
+        TimeoutError(f"Future cancelled after timeout {timeout_s}s"),
+    )
+    # Cancel timeout handler upon completion of the future
+    aio_fut.add_done_callback(lambda _: timeout_handler.cancel())
```

Comment:
I am pretty sure that because the `timeout_handler` does not catch `TimeoutError`, it can log a nasty exception traceback ("unhandled exception in coroutine ..."). Can you add an `except TimeoutError: pass` block to the timeout handler?

Reply:
`timeout_handler` is just a handler for `_try_set_exception` (`_try_cancel_future` previously) why would it need to handle TimeoutError?

Reply:
Oh I'm sorry, typo, I meant `asyncio.CancelledError`